src/core/attributeformmodelbase.cpp

```diff
@@ -239,6 +241,48 @@ void AttributeFormModelBase::updateAttributeValue( QStandardItem *item )
     //set item visibility to false in case it's a linked attribute
     item->setData( !mFeatureModel->data( mFeatureModel->index( fieldIndex ), FeatureModel::LinkedAttribute ).toBool(), AttributeFormModel::CurrentlyVisible );
   }
+  else if ( item->data( AttributeFormModel::ElementType ) == QStringLiteral( "qml" ) ||
+            item->data( AttributeFormModel::ElementType ) == QStringLiteral( "html" ) )
+  {
+    QString code = mEditorWidgetCodes[item];
+
+    QRegularExpression re( "expression\\.evaluate\\(\\s*\\\"(.*?[^\\\\])\\\"\\s*\\)" );
+    QRegularExpressionMatch match = re.match( code );
+    while( match.hasMatch() )
+    {
+      QString expression = match.captured( 1 );
+      expression = expression.replace( QStringLiteral( "\\\"" ), QStringLiteral( "\"" ) );
+
+      QgsExpressionContext expressionContext = mLayer->createExpressionContext();
+      expressionContext.setFeature( mFeatureModel->feature() );
+
+      QgsExpression exp = QgsExpression( expression );
+      exp.prepare( &expressionContext );
+      QVariant result = exp.evaluate( &expressionContext );
+
+      QString resultString;
+      switch( result.type() )
+      {
+        case QMetaType::Int:
+        case QMetaType::UInt:
+        case QMetaType::Double:
+        case QMetaType::LongLong:
+        case QMetaType::ULongLong:
+          resultString = result.toString();
+          break;
+        case QMetaType::Bool:
+          resultString = result.toBool() ? QStringLiteral( "true" ) : QStringLiteral( "false" );
+          break;
+        default:
+          resultString = QStringLiteral( "'%1'" ).arg( result.toString() );
+          break;
+      }
+      code = code.mid( 0, match.capturedStart( 0 ) ) + resultString + code.mid( match.capturedEnd( 0 ) );
+      match = re.match( code );
+    }
+    qDebug() << code;
```

Comment:
remove?

Reply:
Thanks -- there's another debug line I'm removing too (the console.log in FeatureForm.qml)